scripts/renderer.lua

```diff
@@ -79,16 +79,24 @@ function renderer.draw(it, entity_data)
   local is_complex_type = not pipe_types[entity_data.entity.type]
   if is_complex_type then
     local box = flib_bounding_box.resize(entity_data.entity.selection_box, -0.1)
-    entity_data.shape = draw_sprite({
-      sprite = "pv-entity-box",
-      tint = default_color,
-      x_scale = flib_bounding_box.width(box),
-      y_scale = flib_bounding_box.height(box),
-      render_layer = layers.entity,
-      target = flib_bounding_box.center(box),
-      surface = entity_data.entity.surface_index,
-      players = { it.player_index },
-    })
+    local playerSettings = settings.get_player_settings(it.player_index)
```

Comment:
`renderer.draw` is called tens of thousands of times, and Lua will do all of this work every time. Instead of checking these settings here, I would add a boolean the the `Iterator` type that tells the renderer whether or not to draw the box based on the values of the settings when the iterator is created. This way the settings are only indexed once.